compiler/src/dmd/dmangle.d

```diff
@@ -528,7 +528,7 @@ void mangleParameter(Parameter p, ref OutBuffer buf, ref Backref backref)
     if (stc & STC.return_)
         buf.writestring("Nk"); // return
 
-    switch (stc & (STC.IOR | STC.lazy_))
+    switch (stc & ((STC.IOR | STC.lazy_) & ~STC.inscoperef))
```

Comment:
Why?

Reply:
`STC.inscoperef` was added to `STC.IOR`, so negating it keeps the behavior the same (this is a refactoring PR).

Reply:
That sounds wrong.

Reply:
You mean you think that `(STC.IOR | STC.lazy_) & ~STC.inscoperef)` isn't the same as `(STC.IOR | STC.lazy_)` before this PR, or that keeping the behavior the same is wrong?

Reply:
Or perhaps adding `STC.inscoperef` to `STC.IOR` was wrong

Reply:
> Or perhaps adding `STC.inscoperef` to `STC.IOR` was wrong

This. IOR was meant to capture a very common condition as can be seen from the diff. Note that another common condition is `IOR | lazy_`.

Having `inscoperef` being its own thing is IMO not the best idea (I would rather have `inscoperef = STC.in_ | STC.scope_ | STC.ref_ | SOME_UNIQUE_VALUE`) because it creates more invalid states in the code (e.g. if you have `STC.inscoperef` but not `STC.in_`, you have done something wrong somewhere in your code, while you can't do that with the aforementioned declaration).

Reply:
Because STC.constscoperef was added to STC.IOR, so here we subtract it

Reply:
The trouble comes from backwards compatibility. We want to allow in|ref for backwards compatibility, but do not want to allow ref|ref. So simply substituting const|scope|ref would make it impractical to distinguish the cases.